src/adapter/src/coord.rs

```diff
@@ -619,8 +647,11 @@ impl<S: Append + 'static> Coordinator<S> {
                         )])
                         .await
                         .unwrap();
-
-                    policies_to_set.storage_ids.insert(entry.id());
+                    policies_to_set
+                        .entry(policy.expect("sources have a compaction window"))
```

Comment:
I'm having trouble following this, why do all sources, tables, and indexes have an initial compaction window? I thought only those that were considered retained_metrics_relations had initial compaction windows.

Reply:
`initial_logical_compaction_window` always returns `Some` for those types.

If `custom_logical_compaction_window` is not filled in, it just returns the default. It should only return `None` for objects that don't have a notion of compaction window at all (e.g. Views)